### PR TITLE
canonical: Update to `v0.6`

### LINKED ISSUE #41: canonical: Update to `v0.6`

### PR DESCRIPTION
**Pull request by vlopes11**
Canonical 0.6 contains a simplifed interface.

With its update, dusk-plonk brought several API simplifications,
including the ones with the ECC gates.

Resolves #41

### CODE REVIEW

File: CHANGELOG.md
Comment (CPerezz) [inline CHANGELOG.md:12]: Could you add an hyperlink to the issue?

File: Cargo.toml
Comment (CPerezz) [inline Cargo.toml:3]: Don't update the version now please. Do it in a separated commit in the `release-0.7` branch once this gets merged.

**Review by CPerezz — CHANGES_REQUESTED** [review]
A few things to ammend before merging in the comments. Mostly the CHANGELOG stuff and the release bump.

Also, as for the commit message, I would prefer to follow the standards and continue adding the keywords at the beginning of the commit messages unless we're in workspaces.
So I would go from: `canonical: Update to v0.6` to: `Update canonical from v0.5 to v0.6`.

**Comment by ZER0** [comment]
I agreed with all the points mentioned by @CPerezz, therefore I will remove myself from the review so that as soon as they're addressed this PR can be merged (otherwise I will either have to request changes too, and then the PR is blocked twice, or I have to approve it even if want to have the changes applied. And if I will ignore it would still pop up on my reviews queue)

**Review by CPerezz — APPROVED** [review]
LGTM

**Pull request merged**
